### PR TITLE
test and meson cleanups

### LINKED ISSUE #7362: tests fails when systemd.log-level=debug is specified on the kernel command line
```
41/247 hwdb-test                               FAIL     0.47 s
245/247 check-help-udevadm                      FAIL     0.03 s
```
`SYSTEMD_LOG_LEVEL=info ninja -C build test` works OK, so this is not too bad, but it'd be nice to fix at some point.

### PR DESCRIPTION
**Pull request by yuwata**
The tests test-hwdb and check-help-* check the stderr. So, if the `systemd.log_level=debug` is set in the kernel command line, then these tests fail.
This set log_level to info in hwdb-test.sh and meson-check-help.sh, the kernel command line not to change the output of the target programs.

Fixes #7362.

The second commit fixes the recent Fedora CI failure.

### CODE REVIEW

**Comment by yuwata** [comment]
~~@keszybz Fedora CI disappeared...~~

**Comment by yuwata** [comment]
Now building rpm for Fedora rawhide on my local machine succeeds. I hope all the CI pass.

**Comment by yuwata** [comment]
Fedora CI failure seems unrelated.
```
OK:       231
FAIL:       1
SKIP:      18
TIMEOUT:    1
The output from the failed tests:
190/250 test-resolve                            TIMEOUT 30.03 s
```

File: meson.build [hunk 1]
```diff
@@ -442,7 +442,8 @@ foreach ident : ['secure_getenv', '__secure_getenv']
 endforeach
 
 foreach ident : [
-        ['memfd_create',      '''#include <sys/memfd.h>'''],
+        ['memfd_create',      '''#define _GNU_SOURCE
+                                 #include <sys/mman.h>'''],
```
Comment (keszybz) [inline meson.build:446]: Hmm, memfd_create(2) disagrees with glibc apparently. But if is actually in sys/mman.h, that's what we should check. If it turns up in other places in the future, we can just add more header files to the check.

**Comment by keszybz** [comment]
Fedora CI fails with a timeout in test-resolve. Let's merge when Ubuntu passes.

**Comment by yuwata** [comment]
Now all the Ubuntu CI pass. Let's merge this.

File: tools/meson-check-help.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/bin/sh -eu
 
+export SYSTEMD_LOG_LEVEL=info
```
Comment (poettering) [inline tools/meson-check-help.sh:3]: iirc this is a bashism, no? we either should write this as 

```
SYSTEMD_LOG_LEVEL=info
export SYSTEMD_LOG_LEVEL
```

or update the shebang line?

Comment (yuwata) [reply tools/meson-check-help.sh:3]: https://unix.stackexchange.com/questions/193095/where-is-export-var-value-not-available
> It (`export var=22`) is not a bashism but a POSIX compliant syntax. It actually started as a kshism quite a long time ago and was later adopted by almost all Bourne syntax based shells. The only notorious exception is /bin/sh on Solaris 10 and older which sticks to the legacy Bourne shell syntax. Hopefully, Solaris 11 uses a POSIX compliant shell as /bin/sh.

But I do not know whether this is true or not. Where can I find the solid the POSIX compliant syntax?

Comment (keszybz) [reply tools/meson-check-help.sh:3]: I think that's good enough, sounds like it's supported everywhere.

**Pull request merged**